issue writer should serialize anything instead of crashing

right now a logged value that orjson doesn't know how to serialize results in an exception.

We probably want some safe default string representation of these values instead of an exception. We probably don't want to require every log caller to have to think about whether their logged values can be serialized by orjson.

```python
self = <zavod.runtime.issues.DatasetIssues object at 0x11608b0e0>, event = {'dataset': 'testdataset1', 'edge': <E(qux, foo, negative)>, 'event': 'Removing existing edge', 'level': 'warning', ...}

    def write(self, event: Dict[str, Any]) -> None:
...
        record["id"] = hash_data(record)
>       out = orjson.dumps(record, option=orjson.OPT_APPEND_NEWLINE)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       TypeError: Type is not JSON serializable: Edge

zavod/runtime/issues.py:70: TypeError
```

Revert this when fixed
```diff
diff --git a/zavod/zavod/integration/dedupe.py b/zavod/zavod/integration/dedupe.py
index 79a6de540..7b2069598 100644
--- a/zavod/zavod/integration/dedupe.py
+++ b/zavod/zavod/integration/dedupe.py
@@ -106,7 +106,7 @@ def merge_entities(
             edge = resolver.get_resolved_edge(canonical_id, other_id)
             if edge is not None:
                 if force is True or edge.judgement == Judgement.UNSURE:
-                    log.warn("Removing existing edge", edge=edge)
+                    log.warn("Removing existing edge", edge=str(edge))
                     resolver._remove_edge(edge)
                 else:
                     raise ValueError(
```

--- a/zavod/zavod/runtime/issues.py
+++ b/zavod/zavod/runtime/issues.py
@@ -67,7 +67,7 @@ class DatasetIssues(object):
             record["entity"] = {"id": entity}
         record["data"] = data
         record["id"] = hash_data(record)
-        out = orjson.dumps(record, option=orjson.OPT_APPEND_NEWLINE)
+        out = orjson.dumps(record, option=orjson.OPT_APPEND_NEWLINE, default=repr)
         self.fh.write(out)
 
     def clear(self) -> None:

